ddtrace/contrib/pytest/plugin.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+
+from ddtrace import config as ddconfig
+
+from ..trace_utils import int_service
+from ...constants import SPAN_KIND
+from ...ext import SpanTypes, ci, test
+from ...pin import Pin
+from .constants import FRAMEWORK, HELP_MSG, KIND
+
+
+def is_enabled(config):
+    """Check if the ddtrace plugin is enabled."""
+    return config.getoption("ddtrace") or config.getini("ddtrace")
+
+
+def _extract_span(item):
+    """Extract span from `pytest.Item` instance."""
+    return getattr(item, "_datadog_span", None)
+
+
+def _store_span(item, span):
+    """Store span at `pytest.Item` instance."""
+    setattr(item, "_datadog_span", span)
+
+
+def pytest_addoption(parser):
+    """Add ddtrace options."""
+    group = parser.getgroup("ddtrace")
+
+    group._addoption(
+        "--ddtrace",
+        action="store_true",
+        dest="ddtrace",
+        default=False,
+        help=HELP_MSG,
+    )
+
+    parser.addini("ddtrace", HELP_MSG, type="bool")
+
+
+def pytest_configure(config):
+    config.addinivalue_line("markers", "dd_tags(**kwargs): add tags to current span")
+
+    if is_enabled(config):
+        Pin(tags=ci.tags(), _config=ddconfig.pytest).onto(config)
+
+
+def pytest_sessionfinish(session, exitstatus):
+    """Flush open tracer."""
+    pin = Pin.get_from(session.config)
+    if pin is not None:
+        pin.tracer.shutdown()
+
+
+@pytest.fixture(scope="function")
+def ddspan(request):
+    pin = Pin.get_from(request.config)
+    if pin:
+        return _extract_span(request.node)
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_protocol(item, nextitem):
+    pin = Pin.get_from(item.config)
+    if pin is None:
+        yield
+        return
+
+    with pin.tracer.trace(
+        SpanTypes.TEST.value,
```

Comment:
```suggestion
        "pytest.test"
```
would typically be our convention here unless there's particular a reason to use just "test"

Reply:
I have added new configuration for operation name. Let me know what you think about it.